src/debugAdapter/goDebug.ts

```diff
@@ -743,14 +744,34 @@ class GoDebugSession extends LoggingDebugSession {
 			return breakpoints;
 		}).then(breakpoints => {
 			response.body = { breakpoints };
-			this.sendResponse(response);
 			log('SetBreakPointsResponse');
+			return this.sendResponse(response);
 		}, err => {
-			this.sendErrorResponse(response, 2002, 'Failed to set breakpoint: "{e}"', { e: err.toString() });
 			logError(err);
+			return this.sendErrorResponse(response, 2002, 'Failed to set breakpoint: "{e}"', { e: err.toString() });
 		});
 	}
 
+	protected setBreakPointsRequest(response: DebugProtocol.SetBreakpointsResponse, args: DebugProtocol.SetBreakpointsArguments): void {
+		log('SetBreakPointsRequest');
+		if (!this.continueRequestRunning) {
+			this.setBreakPoints(response, args);
+		} else {
+			this.skipStopEventOnce = true;
+			this.delve.callPromise('Command', [{ name: 'halt' }]).then(() => {
+				return this.setBreakPoints(response, args).then(() => {
+					return this.continue();
+				}, err => {
+					logError(err);
+					return this.sendErrorResponse(response, 2009, 'Failed to continue delve after setting the breakpoint: "{e}"', { e: err.toString() });
```

Comment:
Since this error handler is for the continue operation, I would move it up like so:
```
this.delve.callPromise('Command', [{ name: 'halt' }]).then(() => {
   return this.setBreakPoints(response, args).then(() => {
     return this.continue().then(null, err => {
          logError(err);
          this.sendErrorResponse(response, 2009, 'Failed to continue delve after setting the breakpoint: "{e}"', { e: err.toString() });
        })
    }, null);
}, err => {
     logError(err)
     this.sendErrorResponse(response, 2008, 'Failed to halt delve before attempting to set breakpoint: "{e}"', { e: err.toString() });
  });
```

Reply:
Actually, on second thought, `this.continue()` doesnt return a promise.
Also, inside `this.continue()`, if the call to delve failed, it is logged and the error doesnt bubble up here.